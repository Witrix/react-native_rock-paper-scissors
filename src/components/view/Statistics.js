import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    ScrollView,
    Button,
    AsyncStorage
} from 'react-native';
import AppStyle             from '../../themes/styles';
import MyLocalization       from "../../my-lib/MyLocalzation";
import Spacer               from "../general/Spacer";
import StatisticsMode       from "../statistics/StatisticsMode";
import Divider              from "../general/Divider";

export default class Statistics extends Component {
    constructor(props) {
        super(props);
        this.state = {};
    }

    componentDidMount() {
        AsyncStorage.getItem('classic').then(result => {
            let data = JSON.parse(result);
            this.setState({classic: data})
        }).catch(error => {console.warn(error)});
        AsyncStorage.getItem('lizard-spock').then(result => {
            let data = JSON.parse(result);
            this.setState({lizard: data})
        }).catch(error => {console.warn(error)});
    }

    onClickBack() {
        if (this.props.navigation && this.props.navigation.goBack) {
            this.props.navigation.goBack();
            return true;
        }
        return false;
    }

    render() {
        return (
            <View style={style.container}>
                <View style={{flex: 0.9}}>
                    <ScrollView>
                        <Text style={AppStyle.h1}>{MyLocalization.getString('statistics')}</Text>
                        <Spacer size={20}/>
                        <Text style={style.title}>{MyLocalization.getString('modeClassic')} - {MyLocalization.getString('allDifficulties')}</Text>
                        <Spacer size={5}/>
                        <StatisticsMode stat={this.state.classic}/>
                        <Spacer size={20}/>
                        <Divider/>
                        <Spacer size={20}/>
                        <Text style={style.title}>{MyLocalization.getString('modeLizardSpock')} - {MyLocalization.getString('allDifficulties')}</Text>
                        <Spacer size={5}/>
                        <StatisticsMode stat={this.state.lizard}/>
                    </ScrollView>
                </View>
                <View style={{flex: 0.1, marginHorizontal: 10}}>
                    <Button title={MyLocalization.getString('back')} color={"grey"}
                    onPress={this.onClickBack.bind(this)}/>
                </View>
            </View>
        );
    }
}

const style = StyleSheet.create({
    container: {
        flex: 1,
        margin: 10,
    },
    title: {
        fontSize: 18,
        color: 'black'
    },
    text: {
        color: 'black'
    }
});
