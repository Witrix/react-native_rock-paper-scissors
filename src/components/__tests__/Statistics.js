import 'react-native';
import React        from 'react';
import renderer     from 'react-test-renderer';
import Statistics   from "../view/Statistics";

it('Render correctly: No props', () => {
    const tree = renderer.create(
        <Statistics/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: Invalid props', () => {
    const tree = renderer.create(
        <Statistics navigation={"abc"}/>
    );
    expect(tree).toMatchSnapshot();
});
