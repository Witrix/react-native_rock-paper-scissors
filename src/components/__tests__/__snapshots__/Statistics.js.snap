// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly: Invalid props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.9,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Statistics
        </Text>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 18,
            }
          }
        >
          Classic
           - 
          All Difficulties
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            No data available at the moment.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "grey",
              "height": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
            }
          }
        />
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 18,
            }
          }
        >
          Lizard Spock
           - 
          All Difficulties
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            No data available at the moment.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginHorizontal": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "grey",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render correctly: No props 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "margin": 10,
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.9,
      }
    }
  >
    <RCTScrollView>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#000000",
              "fontSize": 25,
              "textAlign": "center",
            }
          }
        >
          Statistics
        </Text>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 18,
            }
          }
        >
          Classic
           - 
          All Difficulties
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            No data available at the moment.
          </Text>
        </View>
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "grey",
              "height": 1,
              "marginHorizontal": 20,
              "marginVertical": 10,
            }
          }
        />
        <View
          style={
            Object {
              "marginTop": 20,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "black",
              "fontSize": 18,
            }
          }
        >
          Lizard Spock
           - 
          All Difficulties
        </Text>
        <View
          style={
            Object {
              "marginTop": 5,
            }
          }
        />
        <View
          style={
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
          >
            No data available at the moment.
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "flex": 0.1,
        "marginHorizontal": 10,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "grey",
              },
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</View>
`;
